content/en/blog/2023/testing-otel-demo/index.md

```diff
@@ -0,0 +1,399 @@
+[Ken Hamric](https://github.com/kdhamric).
+
+The [OpenTelemetry Demo](../../../ecosystem/demo/) is a system that simulates a
+Telescope Shop, consisting of multiple microservices written in different
+languages, each handling a specific capability of this distributed system. Its
+purpose is to demonstrate how [OpenTelemetry](/) tools and SDKs can be used in
+an application to obtain telemetry for monitoring results and even to
+[track problems](../../../docs/demo/#scenarios) across multiple services.
+
+One challenge when maintaining the demo is to add new features to the ecosystem
+and guarantee that the current features and telemetry work as intended. Thinking
+about this issue a few months ago, the OpenTelemetry Demo team
+[started a discussion](https://github.com/open-telemetry/opentelemetry-demo/issues/873)
+to prevent future system changes from having unwanted outcomes to the
+microservices results and its telemetry. This resulted in
+[adding trace-based tests to the demo](https://github.com/open-telemetry/opentelemetry-demo/pull/877).
+
+This article describes how trace-based tests were added to the OpenTelemetry
+demo. It discusses the challenges we faced while building it, the outcomes, and
+how it can help the OpenTelemetry community test the demo and add features with
+greater confidence.
+
+## What is a trace-based test?
+
+Trace-based testing is a type of testing that validates a system's behavior by
+triggering an operation against the system and validating its result by checking
+the system output with the emitted
+[traces](../../../docs/concepts/signals/traces/) generated by the system during
+this call.
+
+This term was popularized by Ted Young in the talk
+[Trace Driven Development: Unifying Testing and Observability](https://www.youtube.com/watch?v=NU-fTr-udZg)
+at KubeCon North America 2018.
+
+To perform a trace-based test, we execute an operation against the system, which
+generates a trace, following these steps:
+
+1. Trigger an operation against the system and collect its output and trace id
+   generated from the operation;
+2. Wait for the system to report the entire trace to the telemetry data store;
+3. Collect the trace data generated by the system during the operation. This
+   data should include timing information, as well as any errors or exceptions
+   encountered.
+4. Validate the operation output and the trace data by checking it against the
+   expected results. This involves analyzing the trace data to ensure the system
+   behaved as expected and the output was correct.
+5. If the trace data does not match the expected results, the test should fail.
+   With the trace data in hand, the developer can investigate the issue and make
+   necessary changes to the system or the test.
+
+![Example of Trace-based test running against a distributed system](./trace-based-testing-diagram.png)
+
+Example of Trace-based test running against a distributed system
```

Comment:
is this intended to be an image description? I think docsy has some capabilities for that.

Reply:
If we cannot have it as an image description, we can just drop this sentence.

Reply:
I've removed the image description entirely.